Release 0.97.2

antoine-de opened the pull request:
Hot fix for navitia.io release.
- Handle multi coverage.
  Big impact on authentication because journeys authentication now handle multi layer of covered regions.
- better handling of reverse proxy
  cherrypick of #387 
- rabbit mq error recovering
  cherrypick of #388

l-vincent-l committed:
Jormungandr: hotfix to make it work when there are 2 coords on coverage

antoine-de committed:
Jormungandr: refactoring journey auth

add a get_region in Journey

and tests added.

still lot's of work to do in authentification process and in the tests

antoine-de committed:
Version 0.97.2

kinnou02 committed:
Add a wsgi middleware for handling https behind a reverse proxy.

The HTTP header X-Scheme can be set for override the scheme use by
jormungandr

kinnou02 committed:
Enabe the passing of headers to que query in integration tests

Add a kwargs parameter on the query function of integration test
framework, this way we can pass headers in the query

kinnou02 committed:
Add test for the ReserseProxied middleware

kinnou02 committed:
Translate comments

Not directly related with this PR

antoine-de committed:
Jormungandr: add log and authentication on resources

antoine-de committed:
jormungandr: authentification is now mandatory

kinnou02 committed:
check if the scheme is one of http or https

antoine-de committed:
Jormungandr: do not check region in no origin

we don't want to odo 'compute_region' if no origin

Commit:
jormungandr: Error handling in stat_manger for RabbitMQ

When rabbitMQ is re-initialized and jormungandr is running,
jormungandr cannot push stat masage to rabbitMQ even if
the connection exists.

If it happens the connection will have to be re-initialized.

antoine-de committed:
jormungandr: quick fix to fix v0

and use jormungandr bdd  instances instead of jormungandr instances
(names not very clear :))

kinnou02 commented on source/jormungandr/jormungandr/authentification.py:
```diff
@@ -50,24 +51,15 @@ def wrapper(*args, **kwargs):
         region = None
         if 'region' in kwargs:
             region = kwargs['region']
+            #TODO revoir comment on gere le lon/lat
```
comme pour le from/to je dirais

antoine-de replied:
donc ce qui est la ca te va ?

kinnou02 commented on source/jormungandr/jormungandr/authentification.py:
```diff
@@ -50,24 +51,15 @@ def wrapper(*args, **kwargs):
-                              (region, region_to))
-            except RegionNotFound:
-                pass
+        #else TODO demander a alex pourquoi on doit dire oui sans region et si ca ne peut pas poser des failles
```
tu dois pouvoir le faire sauter je pense

kinnou02 commented on source/jormungandr/jormungandr/authentification.py:
```diff
@@ -159,8 +149,11 @@ def get_user():
     """
     return the current authenticated User or None
     """
+    logging.warn("get user:")
```
debug

kinnou02 commented on source/jormungandr/jormungandr/authentification.py:
```diff
@@ -159,8 +149,11 @@ def get_user():
         return g.user
     else:
         g.user = User.get_from_token(get_token(), datetime.datetime.now())
+        logging.warn("user = {}".format(g.user))
```
```
logging.warn('user %s', user)
```

antoine-de replied:
squoi l'avantage ? (enfin de tt facon ca va etre benné c'est du debug)

kinnou02 replied:
s'plus court :p 
et logiquement tu peux récupérer les objets dans le handler si tu veu faire des trucs chelou

kinnou02 commented on source/jormungandr/jormungandr/instance_manager.py:
```diff
@@ -158,36 +177,74 @@ def key_of_id(self, object_id):
+            return self.key_of_coord(flon, flat, only_one)
+
+        ptobject = models.PtObject.get_from_uri(object_id)
+        if ptobject:
```
t'es sur que tu veux pas sortir l'authent de instance_manager? au moins avoir une autre méthode qui tiens pas compte des droits, genre on peu continuer a faire vaguement des TU

antoine-de replied:
ben si c'est en public=True ya pas de soucis non ?

tu penses pas que c'est mieux de garder un seul truc ?

kinnou02 replied:
ça à un coté methode qui fait l'café

antoine-de committed:
jormungandr: token mandatory

kinnou02 commented on source/jormungandr/jormungandr/interfaces/v1/Journeys.py:
```diff
@@ -439,10 +443,57 @@ def wrapper(*args, **kwargs):
+        to_regions = set(i_manager.key_of_id(args['destination'], only_one=False))
+
+    if not from_regions:
+        #we didn't get any origin, the region is in the destination's list
```
pas compris

antoine-de replied:
ben on peut avoir que le départ ou que l'arrivée, du coup si on en a qu'un sur les 2 on prend la liste telquelle, sinon on fait l'intersection des 2.

ca te va pas ?

kinnou02 replied:
ben tu peux aussi ne pas avoir trouver de correspondance, donc tu va lancer un itinéraire sur une région ou tu sais deja qu'il n'y a pas de solution

antoine-de replied:
nop si la liste est vide une exception RegionNotFound est jetée.

ca merite peut etre un commentaire effectivement

kinnou02 commented on source/jormungandr/jormungandr/interfaces/v1/Journeys.py:
```diff
@@ -439,10 +443,57 @@ def wrapper(*args, **kwargs):
+        raise RegionNotFound(custom_msg="cannot find a region with {o} and {d} in the same time"
+                             .format(o=args['origin'], d=args['destination']))
+
+    sorted_regions = [r for r in possible_regions]
```
je pense que tu veux:

```
list(possible_regions)
```

antoine-de replied:
vouip

Commit:
jormungandr: Error handling modified after comments

If error of type connection or channel,
reinitialize the connection and if success
Push the stat message.

antoine-de committed:
Jormungandr: corrections after review

Commit:
jormungandr: Error handling in stat_manager corrected.

Message modified
Previous connection released.
